to.etc.domui/src/main/java/to/etc/domui/dom/html/IPageWithDomuiNavigationCheck.java

```diff
@@ -0,0 +1,16 @@
+package to.etc.domui.dom.html;
+
+import org.eclipse.jdt.annotation.NonNull;
+import to.etc.domui.state.UIGotoContext;
+
+/**
+ * Marks the UrlPage for custom handling of leaving the page with unsaved data, caused by domui navigation.
+ */
+public interface IPageWithDomuiNavigationCheck extends IPageWithNavigationCheck {
```

Comment:
I propose to rename that to IPageWithNavigationHandler, because it better specifies what it does. I can do that next merge, no need to do it here.